src/views/AddLocusGeneDiseaseView.vue

```diff
@@ -0,0 +1,397 @@
+            "Unable to submit data. Please try again later.";
+          console.log(error);
+        });
+    },
+  },
+};
+</script>
+<template>
+  <div class="container px-5 py-3" style="min-height: 60vh">
+    <h2>Add New G2P Record</h2>
+    <form
+      class="row g-3 pt-3 pb-1"
+      @submit.prevent="geneSearchBtnClickHandler"
+      v-if="!isSubmitSuccess"
+    >
+      <div class="col-auto">
+        <label for="gene-symbol-input" class="col-form-label">Gene</label>
+      </div>
+      <div class="col-3">
+        <input
+          :class="
+            inputValidation.isLocusValid
+              ? 'form-control'
+              : 'form-control is-invalid'
+          "
+          id="gene-symbol-input"
+          v-model.trim="input.locus"
+          aria-describedby="invalid-gene-symbol-input-feedback"
+        />
+        <div id="invalid-gene-symbol-input-feedback" class="invalid-feedback">
+          Please enter a valid Gene.
+        </div>
+      </div>
+      <div class="col-auto">
+        <button type="submit" class="btn btn-primary mb-3">
+          <i class="bi bi-search"></i> Search
+        </button>
+      </div>
+    </form>
+    <p v-if="!geneData">
+      <i class="bi bi-info-circle"></i> Please enter Gene and click Search to
+      proceed further.
+    </p>
+    <div
+      class="d-flex justify-content-center"
+      v-if="isGeneDataLoading || isSubmitDataLoading"
+      style="margin-top: 250px; margin-bottom: 250px"
+    >
+      <div class="spinner-border text-secondary" role="status">
+        <span class="visually-hidden">Loading...</span>
+      </div>
+    </div>
+    <div class="alert alert-danger mt-3" role="alert" v-if="geneErrorMsg">
+      <div>
+        <i class="bi bi-exclamation-circle-fill"></i> {{ geneErrorMsg }}
+      </div>
+    </div>
+    <div
+      id="curation-form-section"
+      v-if="geneData && !isSubmitDataLoading && !isSubmitSuccess"
+    >
+      <GeneInformation
+        :geneData="geneData"
+        :geneFunctionData="geneFunctionData"
+      />
+      <Publication
+        :fetchPublications="fetchPublications"
+        :publicationsData="publicationsData"
+        :isPublicationsDataLoading="isPublicationsDataLoading"
+        :publicationsErrorMsg="publicationsErrorMsg"
+        @update-publications="(valueArray) => (input.publications = valueArray)"
+      />
+      <ClinicalPhenotype
+        :publicationsData="publicationsData"
+        @update-clinical-phenotype="
+          (valueArray) => (input.phenotypes = valueArray)
+        "
+      />
+      <Genotype
+        :attributesData="attributesData"
+        v-model:allelic-requirement="input.allelic_requirement"
+        @update-cross-cutting-modifiers="
+          (valueArray) => (input.cross_cutting_modifier = valueArray)
+        "
+      />
+      <VariantInformation
+        :publicationsData="publicationsData"
+        @update-variant-types="
+          (valueArray) => (input.variant_types = valueArray)
```

Comment:
If I only select two variant types, for example `frameshift_variant` and `stop_gained`, the variation_types array should only include these two:

```
"variant_types": [
		{
			"comment": "This is a frameshift",
			"de_novo": false,
			"inherited": true,
			"nmd_escape": true,
			"primary_type": "protein_changing",
			"secondary_type": "frameshift_variant",
			"supporting_papers": [
				38737272,
				38768424
			],
			"unknown_inheritence": false
		},
		{
			"comment": "",
			"de_novo": false,
			"inherited": false,
			"nmd_escape": false,
			"primary_type": "protein_changing",
			"secondary_type": "stop_gained",
			"supporting_papers": [
				12451214
			],
			"unknown_inheritence": true
		}
]
```

The code is returning all variant types which is confusing when we select the data to publish.

Reply:
The same applies to `variant_consequences` and `variant_descriptions`.
In my example I only select one variant_consequence (altered_gene_product_structure), this is how the data looks:
```
"variant_consequences": [
		{
			"name": "altered_gene_product_level",
			"support": ""
		},
		{
			"name": "decreased_gene_product_level",
			"support": ""
		},
		{
			"name": "absent_gene_product_level",
			"support": ""
		},
		{
			"name": "increased_gene_product_level",
			"support": ""
		},
		{
			"name": "altered_gene_product_structure",
			"support": "inferred"
		},
		{
			"name": "uncertain",
			"support": ""
		}
	]
```

I would expect the data to be this:
```
"variant_consequences": [
		{
			"name": "altered_gene_product_structure",
			"support": "inferred"
		}
	]
```

Reply:
Using my example for the variant types, I think the data should be like this:
```
"variant_types": [
		{
			"comment": "This is a frameshift",
			"inheritance": "de_novo",
			"nmd_escape": true,
			"primary_type": "protein_changing",
			"secondary_type": "frameshift_variant",
			"supporting_papers": [
				38737272,
				38768424
			]
		},
		{
			"comment": "",
			"inheritance": "unknown",
			"nmd_escape": false,
			"primary_type": "protein_changing",
			"secondary_type": "stop_gained",
			"supporting_papers": [
				12451214
			]
		}
]
```

Reply:
made the change in the latest commit